docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Metrics.adoc

```diff
@@ -76,30 +82,91 @@ If the authentication fails, the  server will reply with a `401 NOT AUTHORIZED`
-be set to `true` to enable them.
-For convenience, WildFly standalone configuration provides expressions to enable the statistics by setting a
-System property `-Dwildfly.statistics-enabled=true` to enable statistics on the subsystems provided by the configuration.
+be set to `true` to enable them. For convenience, WildFly standalone configuration provides expressions to enable the
+statistics by setting a System property `-Dwildfly.statistics-enabled=true` to enable statistics on the subsystems
+provided by the configuration.
+
+== Use in Applications
+
+Unlike the previous metrics system, this new extension exposes an API (that of https://micrometer.io[Micrometer]) to
+applications in order to allow developers to record and export metrics out of the box. To do so, application developers
+will need to inject a `MeterRegistry` instance:
+
+[source,java]
+-----
+package com.redhat.wildfly.micrometerdemo;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import io.micrometer.core.instrument.MeterRegistry;
+
+@RequestScoped
+@Path("/endpoint")
+public class Endpoint {
+    @Inject
+    private MeterRegistry registry;
+
+    @GET
+    public String method() {
+        registry.counter("dummy").increment();
+        return "Counter is " + registry.counter("dummy").count();
+    }
+}
+-----
+
+This provides the application with a `MeterRegistry` instance that will have any recorded metrics exported with the
+system metrics WildFly already exposes. There is no need for an application to include the Micrometer dependencies in the
+application archive, as they are provided by the server out-of-the-box:
+
+[source,xml]
+-----
+<dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-core</artifactId>
+    <version>${version.micrometer}</version>
+    <scope>provided</scope>
+</dependency>
+-----
+
+[[breaking-changes]]
+== Breaking Changes
+
+This new subsystem does bring some breaking changes in order to provide metrics in a manner more consistent with general industry standards.
+
+=== No more prefixes
+This new extension removes the `base` and `vendor` prefixes, in accordance with
+Prometheus conventions. For example, metrics such as `base_classloader_loadedClasses` will now simply be `classloader_loaded_classes`.
+
+Likewise, the `prefix` configuration property has been removed, as the prefixing of meter names is inconsistent with Prometheus best practices. Tags are the preferred way to handle differentiating metrics of the same name.
+
+=== More consistent name formatting
+In previous releases, one might see a meter named `base_classloader_loadedClasses` or `vendor_BufferPool_used_memory`. These have been renamed to `classload_loaded_classes` and `buffer_pool_used_memory` respectively. The "snake casing" of the names should now be consistent across all the meter names.
```

Comment:
@jasondlee Between Beta and Final we need to restore the old file and create a new file for this content, and then tweak as needed to deal with Micrometer only being in WFP.